### ISSUE TITLE
`vm.copy()` doesn't preserve all options, like `allowUnlimitedContractSize`.

### ISSUE THREAD

**Issue by davidmurdoch** [ethereumjs/ethereumjs-monorepo#2017]
I expected initial options to be preserved when calling `vm.copy()`, but this does not seem to be the case:

```javascript
const assert = require("assert");
const VM = require("@ethereumjs/vm").default;

const vm = new VM({
  allowUnlimitedContractSize: true
});
const vmCopy = vm.copy();
assert.strictEqual(vm._allowUnlimitedContractSize, true); // passes
assert.strictEqual(vmCopy._allowUnlimitedContractSize, true); // fails

```

**Comment by holgerd77** [comment]
@davidmurdoch Hi there, yes that's a bug, thanks for reporting!

I guess this is also a candidate which we want to do towards `v5-maintenance` and `master` (currently containing the v5 breaking release beta code), with the `master` changes likely affecting both the `VM` and the `EVM` package.

Here are some references:

- `v5-maintenance` [VM.copy()](https://github.com/ethereumjs/ethereumjs-monorepo/blob/v5-maintenance/packages/vm/src/index.ts#L487) function
- `master` [VM.copy()](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/vm/src/vm.ts#L217) function
- `master` [EVM.copy()](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/evm/src/evm.ts#L910) (there is some opts stuff in there, can't judge on first look, this would at least need a confirmation if things are passed. Current assumption: nevertheless rather not)

I will assign @ScottyPoi here since I think this might fit, Scotty, feel free to unassign if you have other plans or think this is not so fitting.

If you do it would be great if you can add a simple test case (in the form of the code David posted, or at least similar) for every option which needs to be passed.

With these activational options we need to be careful, so `activatePrecompiles` and `activateGenesisState`, so that there is no re-creation/overwriting of state in an already initialized state manager. As an update and when thinking about it: these should not be passed over along with copy().

I guess the very most (all?) of the other options as well, but would be good to give this some additional own thinking. I would think at least the "simple" flag and flag-like options like `hardforkByBlockNumber`, `hardforkByTD`, `allowUnlimitedContractSize`, ... should be safe to pass, also things like the custom opcodes and precompiles (this is actually instead really necessary and would otherwise be a bug as well).

---

@davidmurdoch on other notes: as you could side read we actually have our Beta (1) releases for the upcoming breaking releases out (since last week): https://github.com/ethereumjs/ethereumjs-monorepo/pull/1957

This is a HUGE release round with breaking releases for all libraries and substantial structural changes, if you find some time to test early on that would be really great! 🙂 Feel free to reach out with questions at any time!

**Comment by acolytec3** [comment]
Fixed by #2027
